source/solvers/heat_transfer.cc

```diff
@@ -1082,6 +1081,65 @@ HeatTransfer<dim>::setup_dofs()
                                        &this->previous_solutions);
 }
 
+template <int dim>
+void
+HeatTransfer<dim>::update_boundary_conditions()
+{
+  if (!this->simulation_parameters.boundary_conditions_ht.time_dependent)
+    return;
+
+  double time = this->simulation_control->get_current_time();
+  // We begin by setting the new time for all expressions, although the change
+  // for the convection-radiation boundary conditions won't be applied in this
+  // function
+  for (unsigned int i_bc = 0;
+       i_bc < this->simulation_parameters.boundary_conditions_ht.size;
+       ++i_bc)
+    {
+      this->simulation_parameters.boundary_conditions_ht.value[i_bc]->set_time(
+        time);
+      this->simulation_parameters.boundary_conditions_ht.h[i_bc]->set_time(
+        time);
+      this->simulation_parameters.boundary_conditions_ht.Tinf[i_bc]->set_time(
+        time);
+      this->simulation_parameters.boundary_conditions_ht.emissivity[i_bc]
+        ->set_time(time);
+
+      Assert(
+        this->simulation_parameters.boundary_conditions_ht.emissivity[i_bc]
+              ->value(Point<dim>()) <= 1.0 &&
+          this->simulation_parameters.boundary_conditions_ht.emissivity[i_bc]
+              ->value(Point<dim>()) >= 0.0,
+        EmissivityError(
+          this->simulation_parameters.boundary_conditions_ht.emissivity[i_bc]
+            ->value(Point<dim>())));
+    }
+
+  {
+    nonzero_constraints.clear();
+    nonzero_constraints.reinit(this->locally_relevant_dofs);
+    DoFTools::make_hanging_node_constraints(this->dof_handler,
+                                            nonzero_constraints);
+
+    for (unsigned int i_bc = 0;
+         i_bc < this->simulation_parameters.boundary_conditions_ht.size;
+         ++i_bc)
+      {
+        // Dirichlet condition : imposed temperature at i_bc
+        if (this->simulation_parameters.boundary_conditions_ht.type[i_bc] ==
+            BoundaryConditions::BoundaryType::temperature)
+          {
+            VectorTools::interpolate_boundary_values(
+              this->dof_handler,
+              this->simulation_parameters.boundary_conditions_ht.id[i_bc],
+              *this->simulation_parameters.boundary_conditions_ht.value[i_bc],
+              nonzero_constraints);
+          }
+      }
+  }
```

Comment:
Why do you have this additional scope here?

Reply:
I put it because that's the way that it was written in the setup_dofs functions, but without this additional scope it works just as well so I'll remove it.